### PR TITLE
Display asset size

### PR DESCRIPTION
**Pull request by pevik**
Verification run: http://quasar.suse.cz/tests/7187#downloads
example

Obviously https://github.com/os-autoinst/openQA/pull/4212, i.e. printing "(does not exist)" still works :)
example2

### CODE REVIEW

File: lib/OpenQA/WebAPI/Plugin/Helpers.pm [hunk 1]
```diff
@@ -77,6 +77,24 @@ sub register ($self, $app, $config) {
...
+
+
+            my @units = qw(B KB MB GB TB PB);
+
+            for (@units) {
+                last if $size < 1024;
+                $size /= 1024;
+                $exp++;
+            }
+
```
Comment (okurz) [inline lib/OpenQA/WebAPI/Plugin/Helpers.pm:93]: Please use OpenQA::Utils::human_readable_size

**Review by okurz — DISMISSED** [review]
I like the idea and I think there is no bad impact

File: lib/OpenQA/WebAPI/Plugin/Helpers.pm (hunk 1 above)
Comment (pevik) [reply lib/OpenQA/WebAPI/Plugin/Helpers.pm:93]: Sure, thx for a review.

File: lib/OpenQA/WebAPI/Plugin/Helpers.pm [hunk 2]
```diff
@@ -77,6 +77,12 @@ sub register ($self, $app, $config) {
             return $c->include_branding('external_reporting', %args);
         });
 
+    $app->helper(
+        human_readable_size => sub {
+            my ($c, $size) = @_;
+            return human_readable_size($size);
+        });
```
Comment (Martchus) [inline lib/OpenQA/WebAPI/Plugin/Helpers.pm:84]: Maybe tidy lets you write a one-liner:
```suggestion
    $app->helper(human_readable_size => sub ($c, $size) { human_readable_size $size });
```

Comment (pevik) [reply lib/OpenQA/WebAPI/Plugin/Helpers.pm:84]: It looks it allows it, but there are other functions with just single line which aren't one-liners, thus is it wise to change code style now?

Comment (Martchus) [reply lib/OpenQA/WebAPI/Plugin/Helpers.pm:84]: I would say yes because I'm not a fan of mass changes.

**Review by Martchus — APPROVED** [review]
Also thanks for adding the missing space, it always bugged me.

**Comment by pevik** [comment]
> Also thanks for adding the missing space, it always bugged me. (It might be possible that you need to adapt some tests, though.)

Good point, thx, updated tests in `t/14-grutasks.t`.

File: lib/OpenQA/WebAPI/Plugin/Helpers.pm (hunk 2 above)
Comment (pevik) [reply lib/OpenQA/WebAPI/Plugin/Helpers.pm:84]: I'd prefer to keep the old style (not one liner).

**Comment by pevik** [comment]
Any idea why OBS Package Build failed? I'm looking into log, but not don't understand the output.

**Comment by Martchus** [comment]
That's the relevant output:
```
[  929s] #   Failed test 'Cache limit is default (50GB)'
[  929s] #   at ./t/25-cache.t line 96.
[  929s] #                   '[info] Creating cache directory tree for "/tmp/XMBdF20b9R/t/cache.d/cache"
[  929s] # [info] Cache size of "/tmp/XMBdF20b9R/t/cache.d/cache" is 0 Byte, with limit 50 GiB
[  929s] # '
[  929s] #     doesn't match '(?^u:Cache size of "/tmp/XMBdF20b9R/t/cache.d/cache" is 0 Byte, with limit 50GiB)'
[  929s] 
[  929s] #   Failed test 'Cache limit/size match the expected 100GB/168)'
[  929s] #   at ./t/25-cache.t line 123.
```

So it looks like you just have to insert a space in the expected value of the test check.

**Comment by pevik** [comment]
Ah, I haven't viewed whole file, thanks! Fixed those two.

**Review by mimi1vx — APPROVED** [review]

**Comment by mimi1vx** [comment]
@pevik please fix code stile `make tidy` is your friend

File: t/ui/18-tests-details.t [hunk 3]
```diff
@@ -455,7 +455,7 @@ subtest 'misc details: title, favicon, go back, go to source view, go to log vie
     # way lead to unstability (see poo#94060)
     $driver->get('/tests/99946/downloads_ajax');
     like $driver->find_element_by_id('asset-list')->get_text,
-      qr/openSUSE-13.1-DVD-i586-Build0091-Media.iso[\n|\s]+openSUSE-13.1-x86_64.hda \(does not exist\)/, 'asset list';
+      qr/openSUSE-13.1-DVD-i586-Build0091-Media.iso (0 Byte)[\n|\s]+openSUSE-13.1-x86_64.hda \(does not exist\)/, 'asset list';
```
Comment (Martchus) [inline t/ui/18-tests-details.t:458]: I suppose you need to escape the braces.

Comment (pevik) [reply t/ui/18-tests-details.t]: Thanks, fixed!

**Review by okurz — APPROVED** [review]

**Comment by mimi1vx** [comment]
@pevik last thing ... `tools/tidy` :D

**Pull request merged**